### PR TITLE
BNF-1311 Make left hand menu collapsible on mobile

### PR DESCRIPTION
**Pull request by emollett**
https://nicedigital.atlassian.net/browse/BNF-1311

### CODE REVIEW

File: gatsby/src/components/DetailsPageLayout/DetailsPageLayout.module.scss [hunk 1]
```diff
@@ -5,3 +5,68 @@
...
+  }
+
+  @include core.mq($from: sm) {
+    display: none;
+  }
+}
+
+@include core.mq($from: sm) {
+  :global(.stacked-nav) {
+    display: block !important;
```
Comment (barkertron) [inline gatsby/src/components/DetailsPageLayout/DetailsPageLayout.module.scss:73]: Is there an alternative to using `!important` here - could stack up specificity instead perhaps? Something along the lines of `:global(.stacked-nav.stacked-nav)` might work and would give you the ability to apply further overrides, which `!important` would not.

**Review by barkertron — APPROVED** [review]

File: gatsby/src/pages/medical-devices/{BnfClinicalMedicalDeviceInformationGroup.medicalDeviceType__medicalDevice__slug}/{BnfClinicalMedicalDeviceInformationGroup.slug}.tsx [hunk 2]
```diff
@@ -125,28 +124,15 @@ const CMPIPage: FC<CMPIPageProps> = ({
 			menu={
 				clinicalMedicalDeviceInformationGroups.length > 1
 					? () => (
-							<StackedNav
```
Comment (ediblecode) [inline gatsby/src/pages/medical-devices/{BnfClinicalMedicalDeviceInformationGroup.medicalDeviceType__medicalDevice__slug}/{BnfClinicalMedicalDeviceInformationGroup.slug}.tsx:128]: There's also a stacked nav inside {BnfMedicalDeviceType.slug}.tsx that looks like we've missed

File: gatsby/src/components/DetailsPageLayout/DetailsPageLayout.tsx [hunk 3]
```diff
@@ -98,9 +114,30 @@ export const DetailsPageLayout: React.FC<DetailsPageLayoutProps> = ({
 
 			<Grid gutter="loose" data-testid="body">
 				{Menu && (
-					<GridItem cols={12} md={4} lg={3} className="hide-print">
-						<Menu />
-					</GridItem>
+					<>
+						<GridItem cols={12} md={4} lg={3} className="hide-print">
+							{isClient ? (
```
Comment (ediblecode) [inline gatsby/src/components/DetailsPageLayout/DetailsPageLayout.tsx:119]: I think if we move the logic and styles into the menu component, then it'll be easier to replace all instances with the nds stacked-nav directly if/when we move the logic there.

File: gatsby/src/components/Menu/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { useLocation } from "@reach/router";
+import { Link } from "gatsby";
+import React, { type FC } from "react";
+
+import { StackedNav, StackedNavLink } from "@nice-digital/nds-stacked-nav";
+
+export interface MenuProps {
+	ariaLabel?: string;
```
Comment (ediblecode) [inline gatsby/src/components/Menu/Menu.tsx:8]: If we keep the same prop types as the main stacked nav element then it'll be easier to swap out for the NDS one directly if/when we move the logic into the NDS

**Review by ediblecode — CHANGES_REQUESTED** [review]

**Review by ediblecode — APPROVED** [review]

**Pull request merged**